### PR TITLE
Add Opt to Override Default Header Names

### LINKED ISSUE #7: Add option to set up the name for the headers on config
Instead of predefine the name of the headers, some APIs need to respect some patterns, such as `x-total, x-per-page`.

The ideal scenario would be having the current ones as default and add the possibility to use custom names as config.

### PR DESCRIPTION
**Pull request by kinson**
# Overview

Closes #7 .

Updated the paginate function to let users pass a keyword list of pagination header names to override the default ones.

Thanks to the author of #8 for the inspiration.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -59,6 +59,22 @@ Total: 300
 Per-Page: 10
 ```
 
+### Using Custom Header Names
+
+Override any number of pagination header names by passing opts with a `header_names` keyword list like so:
+
+```elixir
+Screenever.Headers.paginate(page,
+  header_names: [
```
Comment (kinson) [inline README.md:68]: @doomspork I have gone between `header_names`, `headers`, and `header_keys`. What do you think makes the most sense?

Comment (doomspork) [reply README.md:68]: @kinson I think `header_keys` gets my vote on this one 👍

Comment (kinson) [reply README.md]: Makes sense, I just updated the pr! @doomspork

**Review by doomspork — APPROVED** [review]
This great. We need to create an issue to remember to cut a new release, but first make sure there's no other low hanging fruit we could fit in.

**Pull request merged**